server/controllers/blogsController.js

```diff
@@ -1,100 +1,229 @@
-const asyncHandler = require('express-async-handler');
-const Blog = require('../models/blogModel');
+const asyncHandler = require("express-async-handler");
+const mongoose = require("mongoose");
+const Blog = require("../models/blogModel");
 
-// Get all blogs
 exports.getBlogs = asyncHandler(async (req, res) => {
-  const blogs = await Blog.find().sort({ date: -1 });
-  res.json(blogs);
+  try {
+    const blogs = await Blog.aggregate([
+      { $sort: { date: -1 } },
+      {
+        $lookup: {
+          from: "users",
+          localField: "author",
+          foreignField: "_id",
+          as: "authorInfo",
+        },
+      },
+      { $unwind: "$authorInfo" },
+      {
+        $addFields: {
+          authorName: "$authorInfo.fullName",
+        },
+      },
+      {
+        $project: {
+          authorInfo: 0,
+        },
+      },
+    ]);
+
+    res.status(200).json(blogs);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using error handling middleware</b></div>
<div id="fix">

Consider using the error handling middleware instead of directly sending error responses. The error middleware is already set up to handle errors consistently across the application. A similar issue was also found in client/src/pages/HomePage.jsx (line 14-18).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
  } catch (error) {
    throw error;
  }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Divyanshu9822/blog-mern-app/pull/22#issuecomment-2544587479>#245782</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged